Platform/demo/hal/demo_hal.c

```diff
@@ -52,11 +53,14 @@ void HAL_sys_getTime(uint32_t *unix_timestamp) {
 
 SAT_returnState HAL_hk_report(uint8_t sid, void *output) {
   switch (sid) {
-    case BATTERY_1:;
-      HK_battery battery1 = *(HK_battery *)output;
-      HAL_get_current_1(&battery1.current);
-      HAL_get_voltage_1(&battery1.voltage);
-      HAL_get_temperature(&battery1.temperature);
+    case BATTERY_1:
+      if ((sizeof((char *) output) + 1) > csp_buffer_data_size()) {
```

Comment:
@iBacklight At this point we should know for a fact that (sizeof((char *) output) + 1) == csp_buffer_data_size - 2 by definition.
What I'm suggesting is more to make sure that the object we're about to put into the output buffer isn't too big. Something like this:
```
if( sizeof(HK_battery) > csp_buffer_data_size() - 2) // then we should stop
```

Reply:
(2 bytes for subservice, and SID)